packages/flutter/lib/src/material/material_app.dart

```diff
@@ -106,23 +122,32 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
   }
 
   Widget build(BuildContext context) {
+    if (config.onLocaleChanged != null && _localeData == null) {
+      // If the app expects a locale but we don't yet know the locale, then
+      // don't build the widgets now.
+      return new Container();
```

Comment:
Woah. How long are we expecting getting a locale to take? Can't we build with no strings or something, and fill in the strings later? We don't want to be missing any frames in the initial rendering!